lib/openai_agents/models/responses_adapter.ex

```diff
@@ -20,21 +20,49 @@ defmodule OpenAI.Agents.Models.ResponsesAdapter do
     headers = build_headers(config)
     body = Jason.encode!(request)
 
+    # Record generation span for tracing
+    span_id =
+      OpenAI.Agents.Tracing.record_span(
+        :generation,
+        OpenAI.Agents.Tracing.Span.generation_span(
+          request["model"],
+          request,
+          trace_id: Process.get(:current_trace_id)
```

Comment:
Is using a process like this to store the variable safe if you have more than one agent that is invoked from the same process? I wanna make sure it has process safety in a highly concurrent environment where the same main process might be spinning up many agents.